src/Engine/GameAction.java

```diff
@@ -1106,4 +1110,128 @@ public XYCoord getTargetLocation()
       return destination;
     }
   } // ~UnitDeleteAction
+
+  public static class UnitProduceAction implements GameAction
```

Comment:
Missing class comment header/javadoc to match other GameActions.

Reply:
Also, the name is awful close to `UnitProductionAction` - might want to try to differentiate those more.

Reply:
Good point. Should fix that.

Yeah, I noticed the name after I made it... but they're gonna end up being synonyms.
I guess we could do `ProduceUnitAction` and `UnitProduceUnitAction`, heh.

Reply:
I'm starting to feel like GameAction.java is getting a little too big to manage comfortably.
How do you feel about splitting each class out like GameEvent has?

Reply:
Hah. I did those differently partly just to see which one we liked more in the long run. Though they aren't completely comparable. The `UnitModelScheme` subclasses also have many class definitions in one file - what makes `GameAction` more worthy of splitting?

To be clear, I think I'm fine with doing that as long as we organize the files in a sane way.

Reply:
The metric I'm using is "are there enough lines to make it annoying to scroll around and try to find what I'm looking for?"
Also important is how big the individual classes are, though I guess we could break up a file by defining a few well-defined chunks of classes.

DoRUnits and AWBWUnits are both around 800 lines, which is definitely getting unwieldy... but they're not really going to grow and each individual class is pitifully trivial (and having them right next to each other is also nice for comparison and bulk-edit stuff)

UnitActionType is getting up there, at 551 lines, but at least I can still mouse-roller scroll through the whole file pretty quickly.

GameAction is 12k lines, with enough boilerplate per class to make them longer than e.g. `GameInputState`s are.

I think an aggregated "GameEvent.java" with everything in it would be even longer, since we have a number of GameEvents with no associated GameAction.

I think there's also a point where we'd want to start putting groups of files into subfolders (which I ended up doing with the Tester branch's COs because... y'know).
I think GameEvent is pushing towards having enough classes to make that valuable. If I had to pick now, I think I'd do "generic unit" (move/heal/resupply/fight/capture/die/join), "special unit" (flare/repair/produce/transform/load[?]/unload[?]), and "non-unit" (mass damage/weather/defeat/ability/build unit [as a unit isn't the actor]) events.

Reply:
I'm still fine with splitting this up if you care to.